### PR TITLE
Extract trace of ovn bringup

### PR DESCRIPTION
**Pull request by pecameron**
Scan through the generated trace for key lines in ovn bringup and report
them.

SDN-1028 - Trace: generate trace of networking bring up
Trace: generate trace of networking bring up

Signed-off-by: Phil Cameron <pcameron@redhat.com>

### CODE REVIEW

**Comment by pecameron** [comment]
@dcbw PTAL

File: collection-scripts/gather_network_ovn_trace [hunk 1]
```diff
@@ -126,6 +126,45 @@ function gather_service_data {
   PIDS+=($!)
 }
 
+out=${NETWORK_TRACE_PATH}/bringup
+
+# generate a trace of key parts of ovn bringup
+function ovn_startup {
```
Comment (dcbw) [inline collection-scripts/gather_network_ovn_trace:132]: How about `ovn_generate_startup_trace`?

File: collection-scripts/gather_network_ovn_trace [hunk 2]
```diff
@@ -126,6 +126,45 @@ function gather_service_data {
...
+
+	# find the containers that won the election
+	# ovnkube-master
+	grep -e "won leader election" ${tf} | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+	# nbdb and sbdb - the container that starts the election wins
+	grep -e "starting election" ${tf} | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+	# northd - the active container does "Assigned dynamic"
+	grep -e "ovn_northd|INFO|Assigned dynamic" ${tf} | head -1 | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+
+	sort ${out} > ${NETWORK_TRACE_PATH}/ovn-bringup
```
Comment (dcbw) [inline collection-scripts/gather_network_ovn_trace:164]: Same here, let's name the file similar to the function or at least have `trace` somewhere in it

**Comment by dcbw** [comment]
Where in the artifacts does the startup trace show up?

File: collection-scripts/gather_network_ovn_trace (hunk 1 above)
Comment (pecameron) [reply collection-scripts/gather_network_ovn_trace:132]: OK

File: collection-scripts/gather_network_ovn_trace (hunk 2 above)
Comment (pecameron) [reply collection-scripts/gather_network_ovn_trace:164]: ${NETWORK_TRACE_PATH}/ovn-bringup_trace

**Comment by pecameron** [comment]
@dcbw Note the tracing lines in ovs/ovn during bringup are very limited. There is no line that says the daemon is initialized and ready to handle requests. Additional lines during bringup would be very helpful.

**Comment by pecameron** [comment]
@dcbw the bringup trace appears as a peer file of trace. I made the changes, PTAL

File: collection-scripts/gather_network_ovn_trace [hunk 3]
```diff
@@ -126,6 +126,45 @@ function gather_service_data {
...
+
+	# find the containers that won the election
+	# ovnkube-master
+	grep -e "won leader election" ${tf} | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+	# nbdb and sbdb - the container that starts the election wins
+	grep -e "starting election" ${tf} | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+	# northd - the active container does "Assigned dynamic"
+	grep -e "ovn_northd|INFO|Assigned dynamic" ${tf} | head -1 | gawk '{ print $1 " " $2 " " $3 " " $4 "  ACTIVE" }' >> ${out}
+
+	sort ${out} > ${NETWORK_TRACE_PATH}/ovn-bringup_trace
```
Comment (dcbw) [inline collection-scripts/gather_network_ovn_trace:164]: Sorry for my mistype, but `ovn-startup-trace`

Comment (pecameron) [reply collection-scripts/gather_network_ovn_trace]: Changed.

**Comment by pecameron** [comment]
@dcbw PTAL

**Comment by pecameron** [comment]
@sferich888 @dcbw is there anything more to do with this before merging?

**Pull request merged**